### PR TITLE
annotate build pods

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

File: pkg/build/controller/factory/factory.go [hunk 1]
```diff
@@ -218,16 +218,22 @@ type typeBasedFactoryStrategy struct {
...
 	case buildapi.STIBuildStrategyType:
-		return f.STIBuildStrategy.CreateBuildPod(build)
+		pod, err = f.STIBuildStrategy.CreateBuildPod(build)
 	case buildapi.CustomBuildStrategyType:
-		return f.CustomBuildStrategy.CreateBuildPod(build)
+		pod, err = f.CustomBuildStrategy.CreateBuildPod(build)
 	default:
 		return nil, errors.New("No strategy defined for type")
 	}
+	if pod != nil {
```
Comment (liggitt) [inline pkg/build/controller/factory/factory.go]: check `if err != nil` and return early, right?

File: pkg/build/controller/factory/factory.go [hunk 2]
```diff
@@ -218,16 +218,22 @@ type typeBasedFactoryStrategy struct {
...
+		pod.Annotations[buildapi.BuildAnnotation] = build.Name
```
Comment (liggitt) [inline pkg/build/controller/factory/factory.go]: this annotation key is generic enough that I'd be nervous using it as a strong indicator that this pod is itself a build pod (I could see pods getting annotated with the build that spawned them, for example, at some point in the future)

Comment (liggitt) [inline pkg/build/controller/factory/factory.go]: need to allocate the Annotations map if nil:
```
if pod.Annotations == nil {
  pod.Annotations = map[string]string{}
}
```

**Comment by liggitt** [comment]
@spadgett fyi

File: pkg/build/controller/factory/factory.go (hunk 1 above)
Comment (bparees) [inline pkg/build/controller/factory/factory.go]: i debated.  the previous logic returned both args returned from CreateBuildPod, so i was trying to retain that.  Whether i should attempt to set the annotation when doing so or not is debatable.   I figured it's possible "err" is non-fatal and the caller is still going to create the pod, in which case we do want to add the annotation.

File: pkg/build/api/types.go [hunk 3]
```diff
@@ -8,7 +8,10 @@ import (
 )
 
 // BuildLabel is the key of a Pod label whose value is the Name of a Build which is run.
-const BuildLabel = "build"
+const (
+	BuildAnnotation = "build"
```
Comment (liggitt) [inline pkg/build/api/types.go]: we have a mix of namespaced and non-namespaced annotations today... not sure if this should be "openshift.io/build". @smarterclayton, is there a word from on high on annotation key?

File: pkg/build/controller/factory/factory.go (hunk 1 above)
Comment (liggitt) [inline pkg/build/controller/factory/factory.go]: fair enough

**Comment by bparees** [comment]
@smarterclayton @liggitt so for both the key and value, I copied what Deployments are doing:
			Annotations: map[string]string{
				deployapi.DeploymentAnnotation: deployment.Name,
			},

	DeploymentAnnotation = "deployment"

I'm open to alternatives, but I was trying to be consistent.

**Comment by bparees** [comment]
[test]

File: pkg/build/api/types.go (hunk 3 above)
Comment (smarterclayton) [inline pkg/build/api/types.go]: Follow @ironcladlou's examples in his most recent pulls.  Namespaced is required going forward.

Comment (derekwaynecarr) [inline pkg/build/api/types.go]: Are we settled on openshift.io and not openshift.com ?  Would like to rename my finalizer tokens if that is the case.

Comment (liggitt) [inline pkg/build/api/types.go]: that's the etcd namespace we're using, and kubernetes uses kubernetes.io

**Comment by bparees** [comment]
@liggitt @smarterclayton ok, it's "openshift.io/build.name" now.  speak up if you don't like it, otherwise i'll assume acceptance :)

**Comment by smarterclayton** [comment]
Dan I believe is using - on the last part - I don't care about 'build.name' or 'build-name' as long as we have a reason for one or the other and it's internally consistent.  Upstream is using dash for separating words (display-name).  "build.display-name" might be the example of when you would use dash and dot.

----- Original Message -----

**Comment by bparees** [comment]
yeah my impression after looking at dan's and talking to him was to use dash for descriptive field names, but dot to separate fields, so that's what i went with.

**Comment by liggitt** [comment]
I like `build.name`, `foo.display-name` as a pattern

**Comment by liggitt** [comment]
@smarterclayton do we want to update existing annotations to be namespaced and follow that pattern?

**Comment by smarterclayton** [comment]
Ugh.  Timing scares me.

----- Original Message -----

**Comment by smarterclayton** [comment]
So say we all.

----- Original Message -----

**Comment by liggitt** [comment]
@bparees `build.name` is fine for the UI as a signal that this is a build pod for now

**Comment by liggitt** [comment]
LGTM

**Comment by bparees** [comment]
[merge]

**Comment by liggitt** [comment]
Opened #2169 with a list of the annotations the UI is using, just as a reference if we want to change any

**Pull request merged**